File: src/python/aircraft.py
import math, re
import Collision_avoidance.py as Collision_avoidance

class Aircraft:
  neighbors = {}
  
  def __init__(self, myInfo):
    self.ID = myInfo[0]
    self.x = myInfo[1]
    self.y = myInfo[2]
    self.z = myInfo[3]
    self.speed = myInfo[4]
    self.cardinality = myInfo[5]
    
  def listen(self):
    return thing
  
  def broadcast(self):
    return thing
  
  def check_neighbors(self):
    myInfo = [self.ID, self.x, self.y, self.z]
    invInfo = [2,9500,9500,9500]
    caInstance = Collision_avoidance(invInfo,myInfo)
    didAdjust = caInstance.run_ca()
    if (didAdjust>0):
        self.y = caInstance.myY


airplaneInfo = [1,10000,10000,10000,200,40]
print(myairplane.y)
myairplane = Aircraft(airplaneInfo)
myairplane.check_neighbors()
print(myairplane.y)

File: src/version_1/aircraft.cpp
#include <string.h>

#include "aircraft.h"

string Aircraft::broadcast()
{
    //broadcast my current tri_coords
    //go to column on CSV where top box = aircraft ID
      // second box = myCoordiantes.x;
      // third box = myCoordinates.y;
      // fourth box = myCoordinates.z;
      // fifth box = speed;
      // sixth box = cardinality;
}

string Aircraft::check_neighbors()
{
  int numNeighbors, numChecks;
  Collision_avoidance* ca = new Collision_avoidance(myX, myY, myZ, AircraftID);

    // get numNeighbors

  numChecks = numNeighbors;
  while (numChecks >0){
    //For Each Neighbor
    for (int i=0;i<numNeighbors;i++){
      ca.compute_ca();
      numChecks = numChecks - 1;
    }
        //Check CA against neighbor's triCoords
        //if Adjust:
            //Adjust
        //Else
            //do not adjust

  numChecks = numChecks - 1;
  }
}

int Aircraft::listen()
{
    //grab neighbors broadcast info
    // if aircraft nearby, update bool
        //add aircraft to dictionary ifnot already in dictionary
            //dictionary format is {flightID:triCoords}

}

void Aircraft(int inSpeed, int myX, int myY, int myZ, int flightNum)
{
    //initialize all the stuff the aircraft does
    // should have:
        //speed
        //my tri_coordinates
        //flight_ID

}
